mypy/checker.py

```diff
@@ -2506,6 +2506,68 @@ def conditional_type_map(expr: Expression,
         return {}, {}
 
 
+def is_callable_type(type: Optional[Type]) -> bool:
+    """Takes in a type and returns if that type is callable, that is,
+    if `type()` is well-typed (ignoring arguments)."""
+    callable_types = (FunctionLike, AnyType, TypeType)
+    if any(isinstance(type, callable_type) for callable_type in callable_types):
+        return True
+
+    if isinstance(type, UnionType):
+        return all(is_callable_type(item) for item in type.items)
+
+    if isinstance(type, TypeVarType):
+        return not is_callable_type(type.erase_to_union_or_bound())
```

Comment:
You seem to have no tests involving TypeVar. Also none for Type. So please add some!